docs/reference/contributing/target/target.md

```diff
@@ -1,10 +1,46 @@
-<h2 id="target-port">Targets</h2>
+<h2 id="target-port">Porting targets</h2>
 
-Adding a new microcontroller to Arm Mbed OS 5 depends on CMSIS-CORE and CMSIS-Pack. Please make sure that the microcontroller already has these available.
+This document discusses requirements for porting Mbed OS to a derivative target of any one of the Mbed Enabled&trade; targets. To port the MCU to Mbed OS as well, along with a custom board based on the MCU, please follow our [more detailed porting guide](docs/development/reference/porting-targets.html). 
 
-#### Adding a new microcontroller and board
+Adding a new microcontroller to Arm&reg; Mbed OS 5 depends on CMSIS-CORE and CMSIS-Pack. The microcontroller must have these available.
 
-First fork the `mbed-os` repository on GitHub into your own user account. We will use the placeholder `USERNAME` to refer to your username in the following documentation, `MCU_NAME` to refer to the new microcontroller you are adding and `BOARD_NAME` to refer to the new board you are adding. Import an Mbed OS example, and add your fork of `mbed-os` using:
+[Mbed Cloud Client](https://cloud.mbed.com/docs/current) provides reference implementation for three Mbed Enabled targets: the K64F, NUCLEO_F429ZI and the UBLOX_EVK_ODIN_W2. You do not need to port Mbed OS or the Cloud Client if you are using one of these three boards.
+
+### Summary of steps
+
+1. Add your target to `targets.json`.
+   1. Identify your target’s features.
+   1. Identify any existing parent targets.
+   1. Identify macros required for your target's compilation.
+   1. Identify supported compilers and the Mbed OS release version.
+   1. Add the target entry in `targets.json`.
+   1. Verify that you can now pass your target to `mbed compile`.
+1. Add your target port.
+   1. Create the directory structure.
+   1. Add an entry for the target in `mbed_rtx.h`.
+   1. Add startup code, CMSIS headers, NVIC headers and **all** relevant drivers from the CMSIS specifications.
+   1. Implement APIs, HAL, the system clock configuration and any other additional clocks.
+   1. Add toolchain-specific linker descriptions.
+1. Add peripherals and pin names for the target.
+1. Compile with CLI on a supported compiler.
+
+### Target properties
+
+You can identify our sample target by the following:
+
+- Target build name `MY_BOARD_1`.
+- Vendor `MY_VENDOR`, device family `MY_FAMILY` and the device on the board `MY_DEVICE`.
+- A Cortex-M4F based core.
+- Porting for Mbed OS v5.x, which requires support for Arm Compiler 5, GCC Arm Embedded and IAR EWARM.
```

Comment:
This might change, as we are adding ARMv8M platforms which only need to support ARMC6 for now. So, depending on Core Archicture we may support all of these compilers or ARMC6 only if its ARMv8M based core.

Reply:
Agree. Let's add a line here for v8 only requirements.

Reply:
Is that requirement in the right place? Not sure whether you're documenting terminology or requirements